### ISSUE TITLE
Parcel uses relative import path in bundle

### ISSUE THREAD

**Issue by deebov** [parcel-bundler/parcel#4962]
# 🐛 bug report

Parcel is using relative path in bundle

## 🎛 Configuration (.babelrc, package.json, cli command)

`.babelrc`
```js
{
  "presets": ["@babel/preset-typescript"],
  "env": {
    "production": {
      "plugins": ["emotion"]
    },
    "development": {
      "plugins": [["emotion", { "sourceMap": true }]]
    }
  }
}
```

`.parcelrc`
```js
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.{ts,tsx}": ["@parcel/transformer-typescript-tsc"]
  }
}
```

`tsconfig.json`

```js
{
  "compilerOptions": {
    "target": "es5",
    "module": "esnext",
    "lib": ["dom", "esnext"],
    "declaration": true,
    "sourceMap": true,
    "moduleResolution": "node",
    "skipLibCheck": true,
    "strict": true,
    "noFallthroughCasesInSwitch": true,
    "suppressImplicitAnyIndexErrors": true,
    "noImplicitAny": true,
    "strictFunctionTypes": true,
    "strictNullChecks": true,
    "strictPropertyInitialization": true,
    "jsx": "react",
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "downlevelIteration": true,
    "preserveSymlinks": true
  },
  "include": ["packages"],
  "exclude": ["node_modules"]
}
```

## 🤔 Expected Behavior

Parcel should not use relative import path in bundle files.

## 😯 Current Behavior

I am using parcel along with yarn workspaces. In `packages/button/index.tsx` I import a type from `@atomik-ui/system` which is a local package. But in the final `.d.ts` file `@atomik-ui/system` is replaced by  
`../../node_modules/@atomik-ui/system/index`

## 🔦 Context
I am using parcel along with yarn workspaces and typescript to build a design system.

## 💻 Code Sample

final `.d.ts` file:
```ts
import React from "react";
import { BoxProps } from "../../node_modules/@atomik-ui/system/index";
export interface LinkProps extends BoxProps<HTMLAnchorElement> {
    isDisabled?: boolean;
    isExternal?: boolean;
}
export const Link: React.FC<LinkProps>;
```

source file:

```ts
import React, { forwardRef } from "react";
import styled from "@emotion/styled";
import {
  createStyledParser,
  BoxProps,
} from "@atomik-ui/system";

const parser = createStyledParser({
  componentKey: "link",
});

const StyledLink = styled.a<BoxProps>`
  ${parser}
`;

export interface LinkProps extends BoxProps<HTMLAnchorElement> {
  isDisabled?: boolean;
  isExternal?: boolean;
}

export const Link:React.FC<LinkProps> = forwardRef(
   ...
);
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | ^2.0.0-nightly.352
| Node             | v12.18.2
| npm/Yarn         | 1.17.3
| Operating System | macOS 10.15.5

**Comment by deebov** [comment]
if I set `preserveSymlinks` to false in `tsconfig.json` then the path slightly changes to `../system/index` which kinda works but I think it may not be safe to use a relative path

**Comment by devongovett** [comment]
Looks like #3891? That should have been fixed by #4887. Are you on the latest nightly?

**Comment by deebov** [comment]
@devongovett yes, upgrading to the latest nightly solved the issue. Thanks ))
